prow/entrypoint/run.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Run executes the process as configured, writing the output
+// to the process log and the exit code to the marker file on
+// exit.
+func (o Options) Run() error {
```

Comment:
Do we have/want unit tests for any of this? There's actually quite a bit going on here! :)

* error on fail to create process log 
* error and 127 in marker file on fail to start command (is an empty marker file allowed?)
* error if fail to write marker file
* timeout kills process and returns error
* writes the return code to marker file if command complete
* returns error if writing the marker file on completed command fails
* returns nil on success

Reply:
This is a straight copy, we can address in a follow-up.

Reply:
I'd be happy to add more testing here but at the time it was written it was not entirely clear to me how to simply decouple the really low-level actions here (reading and writing a file) from the logic (what is in there?). If we strongly refactored the `Run()` code to use nice `io.Reader`s and what-not I could see some validation of the flow being easier using a test, but generally all of the errors are "shit, the house is on fire" sort of situations so I'm not entirely convinced it's worth it.

Reply:
My concern is we accidentally regress things and now failures are reported as pass or something.

Are you against having a test that actually writes files? Seems like you could hit all these scenarios with a simple bash script or two and hacking up the values for o

Reply:
Yeah, I guess we could have an integration test or tests that interact with the filesystem but I hate `bash` and writing anything DRY or readable or suitable for reuse in it...

Reply:
In any case, since this is a straight move we should handle in a follow-up.

Reply:
You may be able to reuse the scripts we have for this purpose for testing bootstrap